### DISCUSSION TITLE
Continue music on another media player

### DISCUSSION THREAD

**Discussion by shogun160** [music-assistant/support#553] [Feature requests and ideas]
Currently I can't find a way to continue playing music on another  (mass)media_player. The goal I want to achieve is music following me between rooms.
If I start music (through automation or MASS UI) on another media player (in my case all google chromecast audios), the second one will start playing music on top to the first one and is playing the selected playlist from the beginning.

Would be a very nice feature if we could continue playing the current content (song/playlist/podcast/etc.) to make the music follow through different rooms.

**Comment by marcelveldt** [comment]
A plain simple solution to use for now is by using a google cast group in combination with a bunch of automations. I have this myself and I'll share this soon.

**Reply by totalitarian** [reply]
> A plain simple solution to use for now is by using a google cast group in combination with a bunch of automations. I have this myself and I'll share this soon.

Any chance of sharing your setup please

**Comment by shogun160** [comment]
First of all, thanks for your quick answer. Would love to see your code as I didn't manage to do this even by using a google cast group with all available media players.

**Reply by marcelveldt** [reply]
I'll share the blueprints and automations this weekend with some documentation. 
It works pretty well actually. Basically you have a "whole house" cast group where you send the music to.

Each cast player within the group is turned on/off (using mute) and a volume set to the time of day including motion etc.
There's a playlist script to determine what music to play based on the active schedule (time of day)

**Reply by shogun160** [reply]
Ok, got the concept with muting the unused devices. Thanks for the hint. I'll need to make some adjustments to my existing automations as the connected speakers are powered as soon as music starts playing on the google audio device(s).

**Reply by allflame** [reply]
@marcelveldt are you keen on sharing the blueprints / raw automation?

On another note: is there chance to "when the last member of the group will be muted - MA will stop casting" be a switch or a toggle? Maybe I'm doing it wrong (hence the blueprints) but I have rooms where I don't have any speakers, so when I'm muting all those speakers, they won't come back alive cause MA stopped casting.

**Comment by lenuisible** [comment]
Hello there,

I'm also interested in more details about @marcelveldt setup, thanks in advance !

**Comment by OzGav** [comment]
This can be achieved with groups and automations

**Comment by maxi1134** [comment]
I have the manic idea of having music playing from 9 am to 2 am.
Which led me to have to mitigate the annoyance I generated to my 4 immediate neighbors.

I ended up creating 3 automation that mute/unmute and adjust the volume of the rooms depending on their occupancy level.



This one will mute the speakers when a room is empty: https://github.com/maxi1134/Home-Assistant-Config/blob/83797ff853807a328013439ff7d7bad696353579/automations.yaml#L1320


This one should unmute speakers when a room is occupied:
(It will also adjust the volume of the speakers to the average of the "house_speakers" group if the room is unoccupied for longer than 1000 seconds)
https://github.com/maxi1134/Home-Assistant-Config/blob/83797ff853807a328013439ff7d7bad696353579/automations.yaml#L1059


This one will adjust each speaker's volume when it starts playing
(It will not adjust the volume if someone is sleeping in the room and will instead mute it ): https://github.com/maxi1134/Home-Assistant-Config/blob/83797ff853807a328013439ff7d7bad696353579/automations.yaml#L1150


For those too lazy to create the automation!

**Comment by svenvdmeer** [comment]
I would really like for a way to do this in the frontend and not with automations. I often start music in the kitchen and later decide I want to continue listening in the living room. I would be great if there was a clean way to do this other then stopping the music on one player and then start the music on another player. If the music is a radio station it is not a big problem but when i'm halfway through an album I do not know a way to "transfer" the playlist to another player.

**Reply by agdamsbo** [reply]
Create an appropriate player group that includes the players you want to stream to, start the stream to the group player and mute or turn off the players you dont want at the moment. When you change rooms, turn on/off the relevant speaker(s). The stream is on the group entity and will continue. This is the suggested approach in the docs, here: https://music-assistant.io/faq/how-to/

**Reply by OzGav** [reply]
You would do it the same way. Just group your players and mute and unmute the rooms you want.

edit: nice one @agdamsbo you beat me to it!

**Reply by svenvdmeer** [reply]
Thanks for the reaction. That is not really working for me because I dont always want the same music on these players, and i dont always know beforehand if i will want to transfer the  music to another player.

**Reply by OzGav** [reply]
On he flying linking of players is not yet possible but is something likely to be implemented in the future.

**Reply by svenvdmeer** [reply]
No problem In just wanted to clarify my use case.

**Reply by LC7894** [reply]
+1 - would be great of having an easy way of doing this without having to use groups

**Comment by OzGav** [comment]
Transferring the queue is now available in the beta channel with stable to follow soon. You do this from the menu in the top right of the Now Playing view.

**Reply by agdamsbo** [reply]
Just installed the beta to have a go. This is a solid new feature! Grest! Work! How do I best financially support your work?
 saw the project have joined The Open Home Foundation which I think is really great for the community.

**Reply by OzGav** [reply]
Thanks. Have a look here https://music-assistant.io/help/

**Reply by OzGav** [reply]
Transferring the queue via a HA action is now available in integration version 2024.9.1 and above. This functionality is now completed.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
